### PR TITLE
[All end] Update Compose to 1.2 snapshot and other dependencies

### PR DESCRIPTION
**Pull request by florina-muntenescu**

### CODE REVIEW

**Comment by bentrengrove** [comment]
Switched to draft to ensure the snapshot isn't accidentally merged

File: AdvancedStateAndSideEffectsCodelab/app/src/main/java/androidx/compose/samples/crane/base/ExploreSection.kt [hunk 1]
```diff
@@ -75,36 +72,10 @@ fun ExploreSection(
                 style = MaterialTheme.typography.caption.copy(color = crane_caption)
             )
             Spacer(Modifier.height(8.dp))
-            Box(Modifier.weight(1f)) {
```
Comment (simona-anomis) [inline AdvancedStateAndSideEffectsCodelab/app/src/main/java/androidx/compose/samples/crane/base/ExploreSection.kt:78]: FYI, here's the removed bit

File: AdvancedStateAndSideEffectsCodelab/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt [hunk 2]
```diff
@@ -91,57 +87,33 @@ class DetailsActivity : ComponentActivity() {
...
-        uiState.isLoading -> {
-            Box(modifier.fillMaxSize()) {
-                CircularProgressIndicator(
-                    color = MaterialTheme.colors.onSurface,
-                    modifier = Modifier.align(Alignment.Center)
-                )
-            }
-        }
-        else -> { onErrorLoading() }
+    // TODO Codelab: produceState step - Show loading screen while fetching city details
```
Comment (simona-anomis) [inline AdvancedStateAndSideEffectsCodelab/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt:111]: This bit here also seems like a potential rebase issue

**Comment by florina-muntenescu** [comment]
> Switched to draft to ensure the snapshot isn't accidentally merged

This is merging into `1.2-end`, not in `end`. Switching back to PR as it can be merged.

**Pull request merged**